### ISSUE TITLE
pysqlite3-binary

### ISSUE THREAD

**Issue by Loki1920** [shaunthecomputerscientist/EDA-GPT#3]
I'm getting below error when when trying to install 

pip install pysqlite3-binary
ERROR: Could not find a version that satisfies the requirement pysqlite3-binary (from versions: none)
ERROR: No matching distribution found for pysqlite3-binary

how can i fix this..

**Comment by shaunthecomputerscientist** [comment]
> I'm getting below error when when trying to install 
> 
> pip install pysqlite3-binary
> ERROR: Could not find a version that satisfies the requirement pysqlite3-binary (from versions: none)
> ERROR: No matching distribution found for pysqlite3-binary
> 
> 
> how can i fix this..

Did you create a seperate new environment with python > 3.9 and < 3.12? Then installed using requirements.txt?

**Comment by Loki1920** [comment]
yes I have created virtual  environment  ...I'm using Python 3.12.4.....

**Comment by shaunthecomputerscientist** [comment]
> yes I have environment variable ...I'm using Python 3.12.4.....

Okay maybe downgrade it a little bit. Also pysqlite3-binary is used for psycopg2 but i guess you already have sqlite binaries if you are windows user. Try to install other dependencies. And inside Home.py comment out the import of pysqlite3.

**Comment by shaunthecomputerscientist** [comment]
__import__('pysqlite3')
import sys
sys.modules['sqlite3']=sys.modules.pop('pysqlite3')

Only comment these three lines. And see if it works by uploading a dataset. If you don't have api key it will throw error. Make sure you have it.

**Comment by shaunthecomputerscientist** [comment]
Let me know if you face any issues after running it. https://eda-gpt-24.streamlit.app/EDA_GPT I deployed it on streamlit but it's not a powerful server so it might lag. If you have a powerful cpu then it'll work better locally.

**Comment by Loki1920** [comment]
I have already done that  ...but when i run the streamlit ...when i  click the EDA GPT page ...page upload doesn't complete...
![Screenshot 2024-07-26 225615](https://github.com/user-attachments/assets/ea536871-249b-40e4-8e6c-1c20d174ad15)

**Comment by shaunthecomputerscientist** [comment]
Did it show the home page?

**Comment by Loki1920** [comment]
yes ..it is ...

**Comment by shaunthecomputerscientist** [comment]
Show me the logs in the terminal.

**Comment by Loki1920** [comment]
It's not showing any log ...
![image](https://github.com/user-attachments/assets/eae4f41d-78c3-4714-9408-6b9571382dc9)

**Comment by shaunthecomputerscientist** [comment]
Did you reload the page? While it's running?

**Comment by Loki1920** [comment]
yes ..I did but still showing the same thing ...

**Comment by shaunthecomputerscientist** [comment]
![20240726_231159.jpg](https://github.com/user-attachments/assets/4efa429e-1f00-42e9-8d1a-f6db193338d9)

![20240726_231210.jpg](https://github.com/user-attachments/assets/e88fb64a-3876-498c-9b10-368a6bbcc3dd)
This should be logged on initial oage reload. It will download some nltk packages.

**Comment by shaunthecomputerscientist** [comment]
Try to change the page to home then settings and so on. And then come back to EDA page.

**Comment by Loki1920** [comment]
Still facing the same issue ...

**Comment by shaunthecomputerscientist** [comment]
> Still facing the same issue ...

I think it must be some system issue. Since home page is working but for some reason EDA page is not working. Did you try a different browser?

**Comment by Loki1920** [comment]
yes ..I have tried ...but didn't work ...

**Comment by shaunthecomputerscientist** [comment]
> yes ..I have tried ...but didn't work ...

If you want me to make a package which supports structured data analysis i can do that. Or maybe since there are dependency issues i can dockerize it and make a web app for that like Julius ai. The problem is i am not getting time these days. If anyone wants to work on this with me I'll explain the project.
